### PR TITLE
added ide ignore; added super init call for Client; fixed some errors…

### PR DESCRIPTION
**Pull request by zhuyu4839**
added ide ignore; added super init call for Client; fixed some errors for connections.py

### CODE REVIEW

File: udsoncan/client.py [hunk 1]
```diff
@@ -59,6 +59,7 @@ def get_overrided_payload(self, original_payload):
 
 
     def __init__(self, conn, config=default_client_config, request_timeout=None):
+        super().__init__()
```
Comment (pylessard) [inline udsoncan/client.py:62]: Why?

File: udsoncan/connections.py [hunk 2]
```diff
@@ -475,7 +474,8 @@ def close(self):
         self.empty_rxqueue()
         self.empty_txqueue()
         self.exit_requested=True
-        self.rxthread.join()
+        if self.rxthread:
+            self.rxthread.join()
```
Comment (pylessard) [inline udsoncan/connections.py:478]: Thanks

File: udsoncan/client.py (hunk 1 above)
Comment (zhuyu4839) [reply udsoncan/client.py:62]: for subclass inherit.

**Comment by zhuyu4839** [comment]
why failed?

**Pull request closed without merging**